Formula/freetype.rb

```diff
@@ -35,6 +36,7 @@ def install
 
     inreplace [bin/"freetype-config", lib/"pkgconfig/freetype2.pc"],
       prefix, opt_prefix
+    inreplace bin/"freetype-config", Superenv.shims_path, "/opt/homebrew/bin" if OS.mac?
```

Comment:
```
==> brew audit freetype --online --git --skip-style
==> FAILED
Full audit freetype --online --git --skip-style output
  freetype:
    * Files were found with references to the Homebrew shims directory.
      The offending files are:
        bin/freetype-config
  Error: 1 problem in 1 formula detected
```

Hardcoding the default prefix on ARM will not work; that's not necessarily the prefix on Intel macOS, and Linux. Do you have an idea why the shim directory was encoded in `freetype-config`?

Reply:
You are right. The previous fix attempt was wrong. Here I attach the diff of `freetype-config` built without `pkg-config` versus with `pkg-config`.

```diff
@@ -14,7 +14,7 @@
 
 
 # if `pkg-config' is available, use values from `freetype2.pc'
-/opt/homebrew/Library/Homebrew/shims/mac/super/pkg-config --atleast-pkgconfig-version 0.24 >/dev/null 2>&1
+ --atleast-pkgconfig-version 0.24 >/dev/null 2>&1
 if test $? -eq 0 ; then
   # note that option `--variable' is not affected by the
   # PKG_CONFIG_SYSROOT_DIR environment variable
@@ -23,17 +23,17 @@
     export PKG_CONFIG_SYSROOT_DIR
   fi
 
-  prefix=`/opt/homebrew/Library/Homebrew/shims/mac/super/pkg-config --variable prefix freetype2`
-  exec_prefix=`/opt/homebrew/Library/Homebrew/shims/mac/super/pkg-config --variable exec_prefix freetype2`
+  prefix=` --variable prefix freetype2`
+  exec_prefix=` --variable exec_prefix freetype2`
 
-  includedir=`/opt/homebrew/Library/Homebrew/shims/mac/super/pkg-config --variable includedir freetype2`
-  libdir=`/opt/homebrew/Library/Homebrew/shims/mac/super/pkg-config --variable libdir freetype2`
+  includedir=` --variable includedir freetype2`
+  libdir=` --variable libdir freetype2`
 
-  version=`/opt/homebrew/Library/Homebrew/shims/mac/super/pkg-config --modversion freetype2`
+  version=` --modversion freetype2`
 
-  cflags=`/opt/homebrew/Library/Homebrew/shims/mac/super/pkg-config --cflags freetype2`
-  dynamic_libs=`/opt/homebrew/Library/Homebrew/shims/mac/super/pkg-config --libs freetype2`
-  static_libs=`/opt/homebrew/Library/Homebrew/shims/mac/super/pkg-config --static --libs freetype2`
+  cflags=` --cflags freetype2`
+  dynamic_libs=` --libs freetype2`
+  static_libs=` --static --libs freetype2`
 else
   prefix="/opt/homebrew/opt/freetype"
   exec_prefix="/opt/homebrew/opt/freetype"
@@ -45,7 +45,7 @@
 
   cflags="-I${SYSROOT}$includedir/freetype2"
   dynamic_libs="-lfreetype"
-  static_libs="-lfreetype -lz -lbz2 -L/opt/homebrew/Cellar/libpng/1.6.39/lib -lpng16 -lz"
+  static_libs="-lfreetype -lz -lbz2"
   if test "${SYSROOT}$libdir" != "/usr/lib"   &&
      test "${SYSROOT}$libdir" != "/usr/lib64" ; then
     libs_L="-L${SYSROOT}$libdir"
```
It looks like we should just replace the string `/opt/homebrew/Library/Homebrew/shims/mac/super/pkg-config` with `pkg-config`. This way, if `pkg-config` (version > 0.24) is installed and available in `PATH`, `freetype-config` will use it; otherwise `freetype-config` will ignore `pkg-config` and use hardcoded flags. Something like this:
```diff
 
     inreplace [bin/"freetype-config", lib/"pkgconfig/freetype2.pc"],
       prefix, opt_prefix
+    inreplace bin/"freetype-config", Superenv.shims_path/"pkg-config", "pkg-config"
   end
 
   test do
```